subprojects/docs/src/snippets/tasks/incrementalBuild-customTaskClass/groovy/build.gradle

```diff
@@ -114,9 +114,10 @@ tasks.register('packageFiles2', Zip) {
 
 // tag::adhoc-destroyable-task[]
 tasks.register('removeTempDir') {
-    destroyables.register(layout.projectDirectory.dir('tmpDir'))
+    def tempDir = layout.projectDirectory.dir('tmpDir')
+    destroyables.register(tempDir)
     doLast {
-        delete(layout.projectDirectory.dir('tmpDir'))
```

Comment:
🤔 I think we should prefer `FileOperations#delete` over APIs.

Reply:
Thanks, I've added this to https://github.com/gradle/gradle/issues/22536#issuecomment-1342823682. It isn't really feasible prior to having getters for injectable services.